### PR TITLE
Add build version to the installer

### PR DESCRIPTION
**Pull request by mattock**
This allows distinguishing between installers that share the same
driver code, but which have different installers, documentation or
such.

Signed-off-by: Samuli Seppänen <samuli@openvpn.net>

### CODE REVIEW

File: installer/tap-windows6.nsi [hunk 1]
```diff
@@ -38,9 +38,9 @@ InstallDirRegKey HKLM "SOFTWARE\${PRODUCT_NAME}" ""
 ;--------------------------------
 ;Modern UI Configuration
 
-Name "${PRODUCT_NAME} ${PRODUCT_VERSION}"
+Name "${PRODUCT_NAME} ${PRODUCT_VERSION}-${PRODUCT_TAP_WIN_BUILD}"
 
-!define MUI_WELCOMEPAGE_TEXT "This wizard will guide you through the installation of ${PRODUCT_NAME}, a kernel driver to provide virtual tap device functionality on Windows originally written by James Yonan.\r\n\r\nNote that the Windows version of ${PRODUCT_NAME} will only run on Windows XP or later.\r\n\r\n\r\n"
+!define MUI_WELCOMEPAGE_TEXT "This wizard will guide you through the installation of ${PRODUCT_NAME}, a kernel driver to provide virtual tap device functionality on Windows originally written by James Yonan.\r\n\r\nNote that ${PRODUCT_NAME} will only run on Windows Vista or later.\r\n\r\n\r\n"
```
Comment (mattock) [inline installer/tap-windows6.nsi]: This change does not really belong to this commit, so if you want, I can make it separate. In any case the original wording was confusing as it was copy-and-paste from the OpenVPN installer.

File: version.m4 [hunk 2]
```diff
@@ -5,6 +5,8 @@ define([PRODUCT_VERSION_RESOURCE], [9,0,0,21])
 define([PRODUCT_TAP_WIN_COMPONENT_ID], [tap0901])
 define([PRODUCT_TAP_WIN_MAJOR], [9])
 define([PRODUCT_TAP_WIN_MINOR], [21])
+define([PRODUCT_TAP_WIN_REVISION], [2])
```
Comment (mattock) [inline version.m4]: This variable is not used for anything right now, but I think it should be here for clarity.

**Comment by mattock** [comment]
Prior to this change the installer file would be named as _tap-windows-9.21.2.exe_. With this PR the file will be called _tap-windows-9.21.2-I001_ instead. This is in line with how openvpn-build/windows-nsis works.

**Comment by selvanair** [comment]
I do not have a setup to test the build, but these changes look sane.
ACK.

**Comment by mattock** [comment]
@selvanair : there's a test installer [here](https://build.openvpn.net/downloads/temp/tap-windows-9.21.2-I001.exe) if you want to have a look.

One question, though: right now PRODUCT_TAP_WIN_BUILD is set to _001_. As discussed [here](https://community.openvpn.net/openvpn/wiki/GettingTapWindows) our tap-windows driver naming scheme is confusing enough as-is. I think we should consider making the default build number _601_ to match the numbering scheme in OpenVPN installers. Does that sound ok? If yes, I will squash the commits, change the build number and merge this PR.

**Comment by selvanair** [comment]
@mattock: you know best about the numbering scheme so leave it your judgement.. 601 instead of 001 looks OK to me. The driver version numbers (9.9 vs 9.21 which shows up as 9.0.0.21) are indeed totally confusing.

**Comment by mattock** [comment]
I squashed the commits and made build number default to "601" and will now merge this PR.

@selvanair : Can you think of any why we could not change PRODUCT_VERSION_RESOURCE to 9.21.2.601 in version.m4?

**Comment by selvanair** [comment]
You mean 9,21,2,601, right? The version resource is supposed to be major,minor,build,revision -- at least major and minor are consistently interpreted (9 and 21 in our case) tap sources and openvpn. As for 2  I am not sure whether its build or revision.

External programs gets the version from the driver using an IOCTL and that returns three numbers: major (9), minor (21) and 0 or 1 depending debug is enabled or not, in that order. So that is consistent with 9.21.x.x as well.

That said I think this definition of PRODUCT_VERSION_RESOURCE is not used anywhere. The driver gets its version number most likely from this line in the OemVista.inf.in

DriverVer=@PRODUCT_TAP_WIN_RELDATE@,@PRODUCT_TAP_WIN_MAJOR@.00.00.@PRODUCT_TAP_WIN_MINOR@

Also src/resource.rc has

#define VER_PRODUCTVERSION	 PRODUCT_TAP_WIN_MAJOR,00,00,PRODUCT_TAP_WIN_MINOR

I think changing these to major,minor,x,y along with a updated release date should be safe. If you generate one I can test it.

**Pull request merged**